Sources/ClientRuntime/Networking/Http/Middlewares/RequestBody/PayloadBodyMiddleware.swift

```diff
@@ -6,27 +6,27 @@
 //
 
 import struct Foundation.Data
-import typealias SmithyReadWrite.DocumentWritingClosure
+import protocol SmithyReadWrite.SmithyWriter
 import typealias SmithyReadWrite.WritingClosure
 
 public struct PayloadBodyMiddleware<OperationStackInput,
                                     OperationStackOutput,
                                     OperationStackInputPayload,
-                                    Writer>: Middleware {
+                                    Writer: SmithyWriter>: Middleware {
     public let id: Swift.String = "PayloadBodyMiddleware"
 
-    let documentWritingClosure: DocumentWritingClosure<OperationStackInputPayload, Writer>
+    let rootNodeInfo: Writer.NodeInfo
```

Comment:
Like in BodyMiddleware above, a document writing closure is not needed but root node info still is.